cmd/uptrace/main.go

```diff
@@ -282,26 +261,25 @@ func runPGMigrations(ctx context.Context, logger *otelzap.Logger, pg *bun.DB) er
 }
 
 func initClickhouse(lc fx.Lifecycle, logger *otelzap.Logger, conf *bunconf.Config, pg *bun.DB, ch *ch.DB) {
-	lc.Append(fx.Hook{
-		OnStart: func(ctx context.Context) error {
-			if err := ch.Ping(ctx); err != nil {
-				return fmt.Errorf("ClickHouse Ping failed: %w", err)
-			}
+	lc.Append(fx.StartHook(func(ctx context.Context) error {
+		if err := ch.Ping(ctx); err != nil {
+			return fmt.Errorf("ClickHouse Ping failed: %w", err)
+		}
 
-			if err := bunapp.WithGlobalLock(ctx, pg, func() error {
-				return runCHMigrations(ctx, logger, conf, ch)
-			}); err != nil {
+		if err := bunapp.WithGlobalLock(ctx, pg, func() error {
+			return runCHMigrations(ctx, logger, conf, ch)
```

Comment:
Looks like now migrations can't take longer than 15 seconds (StartTimeout) from context.Context. Some migrations can definitely take longer than that.

Let not use `StartHook` to run migrations. Just run them directly in `Invoke` without a timeout.